File: ScorePulse/Models/Bar.swift
import Foundation

/// Represents a bar with a time signature change
/// Only bars with time signature changes need to be stored
struct Bar: Codable, Identifiable {
    let id: UUID
    let number: Int
    let timeSignature: TimeSignature
    
    init(id: UUID = UUID(), number: Int, timeSignature: TimeSignature) {
        self.id = id
        self.number = number
        self.timeSignature = timeSignature
    }
}

File: ScorePulseAndroid/app/src/main/java/com/scorepulse/model/Bar.kt
package com.scorepulse.model

import kotlinx.serialization.Serializable
import java.util.UUID

/**
 * Represents a bar with a time signature change.
 * Only bars with time signature changes need to be stored.
 */
@Serializable
data class Bar(
    @Serializable(with = UUIDSerializer::class)
    val id: UUID = UUID.randomUUID(),
    val number: Int,
    val timeSignature: TimeSignature
)
